src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -572,5 +601,27 @@ public void Dispose()
                 }
             }
         }
+
+        private static string GetMSbuild(string msbuildDirectory)
```

Comment:
Maybe you could also look for the MSBuild/xbuild launch scripts in $PATH? This would also work on Linux when MSBuild is available there.

```csharp
string[] pathDirs = Environment
	.GetEnvironmentVariable("PATH")
	?.Split(new[] { Path.PathSeparator }, StringSplitOptions.RemoveEmptyEntries);

if (pathDirs != null)
{
	foreach (var exeName in new[] { "msbuild", "xbuild" })
	{
		var exePath = pathDirs.Select(dir => Path.Combine(dir, exeName)).FirstOrDefault(File.Exists);
		if (exePath != null)
		{
			return exePath;
		}
	}
}
```